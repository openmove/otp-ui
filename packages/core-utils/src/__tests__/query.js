import {
  restoreDateNowBehavior,
  setDefaultTestTime
} from "../../../../test-utils/time";

import { getDefaultQuery, planParamsToQuery } from "../query";

describe("query", () => {
  afterEach(restoreDateNowBehavior);

  describe("planParamsToQuery", () => {
    it("should parse a depart at query", async () => {
      expect(
        await planParamsToQuery({
          arriveBy: "false",
          bannedRoutes: "897ABC",
          companies: "",
          date: "2019-10-31",
          fromPlace:
            "Guide Dogs for the Blind, Portland, OR, USA::45.517373,-122.675601",
          ignoreRealtimeUpdates: "true",
          maxWalkDistance: "1207",
          mode: "BUS,TRAM,RAIL,GONDOLA,WALK",
          optimize: "QUICK",
          showIntermediateStops: "true",
          time: "17:45",
          toPlace: "Weather Machine, Portland, OR, USA::45.519015,-122.679321",
          ui_activeItinerary: "0",
          ui_activeSearch: "irc7h4rb8",
          walkSpeed: "1.34"
        })
      ).toMatchSnapshot();
    });
  });

  describe("getDefaultQuery", () => {
    it("should return default query", () => {
      setDefaultTestTime();
      expect(getDefaultQuery()).toMatchSnapshot();
    });
  });
});
