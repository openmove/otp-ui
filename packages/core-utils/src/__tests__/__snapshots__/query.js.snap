// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query getDefaultQuery should return default query 1`] = `
Object {
  "bannedRoutes": "",
  "bikeSpeed": 3.58,
  "companies": null,
  "date": "2019-08-04",
  "departArrive": "NOW",
  "endTime": "09:00",
  "from": null,
  "ignoreRealtimeUpdates": false,
  "intermediatePlaces": Array [],
  "maxBikeDistance": 4828,
  "maxBikeTime": 20,
  "maxEScooterDistance": 4828,
  "maxWalkDistance": 1207,
  "maxWalkTime": 15,
  "mode": "WALK,TRANSIT",
  "numItineraries": 3,
  "optimize": "QUICK",
  "optimizeBike": "SAFE",
  "otherThanPreferredRoutesPenalty": 900,
  "preferredRoutes": "",
  "routingType": "ITINERARY",
  "showIntermediateStops": true,
  "startTime": "07:00",
  "time": "19:34",
  "to": null,
  "walkSpeed": 1.34,
  "watts": 250,
  "wheelchair": false,
}
`;

exports[`query planParamsToQuery should parse a depart at query 1`] = `
Object {
  "bannedRoutes": "897ABC",
  "companies": "",
  "date": "2019-10-31",
  "departArrive": "DEPART",
  "from": Object {
    "lat": 45.517373,
    "lon": -122.675601,
    "name": "Guide Dogs for the Blind, Portland, OR, USA",
  },
  "ignoreRealtimeUpdates": "true",
  "maxWalkDistance": 1207,
  "mode": "BUS,TRAM,RAIL,GONDOLA,WALK",
  "optimize": "QUICK",
  "showIntermediateStops": "true",
  "time": "17:45",
  "to": Object {
    "lat": 45.519015,
    "lon": -122.679321,
    "name": "Weather Machine, Portland, OR, USA",
  },
  "ui_activeItinerary": 0,
  "ui_activeSearch": "irc7h4rb8",
  "walkSpeed": 1.34,
}
`;
